### ISSUE TITLE
Error an illegal memory access was encountered at line 90 in file /src/csrc/ops.cu

### ISSUE THREAD

**Issue by duarteocarmo** [axolotl-ai-cloud/axolotl#1467]
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/OpenAccess-AI-Collective/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

Normal inference

### Current behaviour

When trying to infer, an error message happens, `Error an illegal memory access was encountered at line 90 in file /src/csrc/ops.cu`

### Steps to reproduce

1. Build image based on `winglian/axolotl-cloud:main-latest`
2. Run `accelerate launch -m axolotl.cli.train config.yml`
3. Run inference with `python -m axolotl.cli.inference config.yml  --lora_model_dir="./out"`

### Config yaml

```yaml
base_model: google/gemma-2b
model_type: AutoModelForCausalLM
tokenizer_type: AutoTokenizer

load_in_8bit: false
load_in_4bit: true
strict: false

datasets:
  - path: duarteocarmo/lusiadas
    type: alpaca
hub_model_id: duarteocarmo/lusiaidas
val_set_size: 0.1
output_dir: ./out

adapter: qlora
lora_r: 32
lora_alpha: 16
lora_dropout: 0.05
lora_target_linear: true

sequence_len: 4096
sample_packing: true
eval_sample_packing: false
pad_to_sequence_len: true

wandb_project: lusiaidas
wandb_entity:
wandb_watch:
wandb_name:
wandb_log_model: checkpoint


gradient_accumulation_steps: 3
micro_batch_size: 2
num_epochs: 4
optimizer: adamw_bnb_8bit
lr_scheduler: cosine
learning_rate: 0.0002

train_on_inputs: false
group_by_length: false
bf16: auto
fp16:
tf32: false

gradient_checkpointing: true
early_stopping_patience:
resume_from_checkpoint:
local_rank:
logging_steps: 1
xformers_attention:
flash_attention: true

warmup_ratio: 0.1
evals_per_epoch: 4
eval_table_size:
eval_max_new_tokens: 128
saves_per_epoch: 1
debug:
deepspeed:
weight_decay: 0.0
fsdp:
fsdp_config:
special_tokens:
```

### Possible solution

Training goes well, no problem. model gets uploaded to HF. 

However, trying to inference after: 

```
root@C.10334496:/workspace/axolotl$ python -m axolotl.cli.inference config.yml  --lora_model_dir="./out"
[2024-03-30 19:28:43,374] [INFO] [datasets.<module>:58] [PID:1558] PyTorch version 2.1.2+cu118 available.
[2024-03-30 19:28:44,562] [INFO] [real_accelerator.py:191:get_accelerator] Setting ds_accelerator to cuda (auto detect)
                                 dP            dP   dP
                                 88            88   88
      .d8888b. dP.  .dP .d8888b. 88 .d8888b. d8888P 88
      88'  `88  `8bd8'  88'  `88 88 88'  `88   88   88
      88.  .88  .d88b.  88.  .88 88 88.  .88   88   88
      `88888P8 dP'  `dP `88888P' dP `88888P'   dP   dP



[2024-03-30 19:28:45,776] [DEBUG] [axolotl.normalize_config:79] [PID:1558] [RANK:0] bf16 support detected, enabling for this configuration.
[2024-03-30 19:28:46,092] [INFO] [axolotl.normalize_config:182] [PID:1558] [RANK:0] GPU memory usage baseline: 0.000GB (+0.456GB misc)
[2024-03-30 19:28:46,093] [INFO] [axolotl.common.cli.load_model_and_tokenizer:50] [PID:1558] [RANK:0] loading tokenizer... google/gemma-2b
[2024-03-30 19:28:47,332] [DEBUG] [axolotl.load_tokenizer:250] [PID:1558] [RANK:0] EOS: 1 / <eos>
[2024-03-30 19:28:47,332] [DEBUG] [axolotl.load_tokenizer:251] [PID:1558] [RANK:0] BOS: 2 / <bos>
[2024-03-30 19:28:47,332] [DEBUG] [axolotl.load_tokenizer:252] [PID:1558] [RANK:0] PAD: 0 / <pad>
[2024-03-30 19:28:47,332] [DEBUG] [axolotl.load_tokenizer:253] [PID:1558] [RANK:0] UNK: 3 / <unk>
[2024-03-30 19:28:47,332] [INFO] [axolotl.load_tokenizer:264] [PID:1558] [RANK:0] No Chat template selected. Consider adding a chat template for easier inference.
[2024-03-30 19:28:47,332] [INFO] [axolotl.common.cli.load_model_and_tokenizer:52] [PID:1558] [RANK:0] loading model and (optionally) peft_config...
Gemma's activation function should be approximate GeLU and not exact GeLU.
Changing the activation function to `gelu_pytorch_tanh`.if you want to use the legacy `gelu`, edit the `model.config` to set `hidden_activation=gelu`   instead of `hidden_act`. See https://github.com/huggingface/transformers/pull/29402 for more details.
[2024-03-30 19:28:50,461] [INFO] [accelerate.utils.modeling.get_balanced_memory:965] [PID:1558] We will use 90% of the memory on device 0 for storing the model, and 10% for the buffer to avoid OOM. You can set `max_memory` in to a higher value to use more memory (at your own risk).
Loading checkpoint shards: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:01<00:00,  1.07it/s]
[2024-03-30 19:28:53,072] [INFO] [axolotl.load_model:651] [PID:1558] [RANK:0] GPU memory usage after model load: 1.929GB (+0.005GB cache, +1.446GB misc)
[2024-03-30 19:28:53,080] [INFO] [axolotl.load_model:697] [PID:1558] [RANK:0] converting PEFT model w/ prepare_model_for_kbit_training
[2024-03-30 19:28:53,082] [INFO] [axolotl.load_model:706] [PID:1558] [RANK:0] converting modules to torch.bfloat16 for flash attention
[2024-03-30 19:28:53,084] [INFO] [axolotl.load_lora:850] [PID:1558] [RANK:0] found linear modules: ['gate_proj', 'v_proj', 'q_proj', 'down_proj', 'k_proj', 'up_proj', 'o_proj']
[2024-03-30 19:28:53,084] [DEBUG] [axolotl.load_lora:887] [PID:1558] [RANK:0] Loading pretrained PEFT - LoRA
trainable params: 39,223,296 || all params: 2,545,395,712 || trainable%: 1.540950816216351
[2024-03-30 19:28:54,700] [INFO] [axolotl.load_model:751] [PID:1558] [RANK:0] GPU memory usage after adapters: 2.075GB (+1.894GB cache, +1.446GB misc)
================================================================================
Give me an instruction (Ctrl + D to submit):
Below is an instruction that describes a task. Write a response that appropriately completes the request.

### Instruction:
Escreve uma estrofe ao estilo de Os Lusíadas

### Response:========================================
<bos>Below is an instruction that describes a task. Write a response that appropriately completes the request.

### Instruction:
Escreve uma estrofe ao estilo de Os Lusíadas

### Error an illegal memory access was encountered at line 90 in file /src/csrc/ops.cu

```

### Which Operating Systems are you using?

- [X] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.10.14

### axolotl branch-commit

winglian/axolotl-cloud:main-latest

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of axolotl.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by nampdn** [comment]
I can confirm that this is the problem with docker latest image https://hub.docker.com/layers/winglian/axolotl-runpod/main-latest/images/sha256-e476401ab682bd0e3b5d1858679e6a986185ef4cba69ad0242d4ab953607e8da?context=explore
